pyaedt/emit_core/results/results.py

```diff
@@ -140,44 +140,37 @@ def revision_names(self):
         """
         return [rev.name for rev in self.revisions]
 
-    @pyaedt_function_handler()
-    def analyze(self, revision_name=None):
+    @pyaedt_function_handler
+    def get_revision(self, revision_name=None):
         """
-        Analyze the specified design.
+        Load the specified revision.
 
         Parameters
         ----------
         revision_name : str
-            Revision to analyze. The default is ``None``,in which case the most recent revision
-            is loaded if it matches the current design revision.
+            Revision to load. If revision_name = None, the
+            latest revision will be returned.
 
         Returns
         -------
         rev:class:`pyaedt.modules.Revision`
-            Specified ``Revision`` object that was generated.
+            Specified ``Revision`` object that was loaded.
 
         Examples
         --------
-        >>> rev = aedtapp.results.analyze()
+        >>> rev = aedtapp.results.get_revision("Revision 15")
         >>> interferers = rev.get_interferer_names()
         >>> receivers = rev.get_receiver_names()
         """
+        # no revisions to load, create a new one
+        if len(self.revisions) == 0:
+            return self.analyze()
+        # retrieve the latest revision if nothing specified
         if revision_name is None:
-            # analyze the current design revision
-            if self.current_revision is None:
-                self.current_revision = self._add_revision()
-            elif (
-                self.current_revision.revision_number
-                == self.emit_project.odesktop.GetActiveProject().GetActiveDesign().GetRevision()
-            ):
-                # Revision exists for design rev #, load if it needed
-                if not self.current_revision.revision_loaded:
-                    self.current_revision._load_revision()
-            else:
-                # there are changes since the current revision was analyzed, create
-                # a new revision
-                self.current_revision.revision_loaded = False
-                self.current_revision = self._add_revision()
+            # unload the current revision and load the latest
+            self.current_revision.revision_loaded = False
+            self.current_revision = self.revisions[-1]
+            self.current_revision._load_revision()
```

Comment:
@jsalant22 One thing I noticed is that _load_revision() prints a message if it is already loaded.  Doing that the way we are using _load_revision() here seems like a bad idea--depending on what the user is doing it could print that out a lot.  Do you think maybe we should get rid of that print statement?

Reply:
Might work to swap out the print for a logging.debug().  Seems like something we might want to start using.